lib/uffizzi/cli/preview.rb

```diff
@@ -0,0 +1,268 @@
+        when 'failed'
+          spinner.error
+        end
+      end
+    end
+
+    def handle_delete_command(deployment, project_slug)
+      return Uffizzi.ui.say("Preview should be specified in 'deployment-PREVIEW_ID' format") unless deployment_name_valid?(deployment)
+
+      hostname = ConfigFile.read_option(:hostname)
+      deployment_id = deployment.split('-').last
+      params = { id: deployment_id }
+
+      response = delete_deployment(hostname, project_slug, deployment_id, params)
+
+      if ResponseHelper.no_content?(response)
+        handle_succeed_delete_response(deployment_id)
+      else
+        ResponseHelper.handle_failed_response(response)
+      end
+    end
+
+    def handle_describe_command(deployment, project_slug)
+      return Uffizzi.ui.say("Preview should be specified in 'deployment-PREVIEW_ID' format") unless deployment_name_valid?(deployment)
+
+      hostname = ConfigFile.read_option(:hostname)
+      deployment_id = deployment.split('-').last
+      params = { id: deployment_id }
+
+      response = describe_deployment(hostname, project_slug, deployment_id, params)
+
+      if ResponseHelper.ok?(response)
+        handle_succeed_describe_response(response)
+      else
+        ResponseHelper.handle_failed_response(response)
+      end
+    end
+
+    def handle_activity_items_response(response)
+      unless ResponseHelper.ok?(response)
+        @spinner.error
+        ResponseHelper.handle_failed_response(response)
+      end
+    end
+
+    def handle_succeed_list_response(response)
+      deployments = response[:body][:deployments]
+      deployments.each do |deployment|
+        Uffizzi.ui.say("deployment-#{deployment[:id]}")
+      end
+    end
+
+    def handle_succeed_delete_response(deployment_id)
+      Uffizzi.ui.say("Preview deployment-#{deployment_id} deleted")
+    end
+
+    def handle_succeed_describe_response(response)
+      deployment = response[:body][:deployment]
+      deployment.each_key do |key|
+        Uffizzi.ui.say("#{key}: #{deployment[key]}")
+      end
+    end
+
+    def prepare_params(file_path)
+      begin
+        compose_file_data = File.read(file_path)
+      rescue Errno::ENOENT => e
+        Uffizzi.ui.say(e)
+        return
+      end
+
+      compose_file_dir = File.dirname(file_path)
+      dependencies = ComposeFileService.parse(compose_file_data, compose_file_dir)
+      absolute_path = File.absolute_path(file_path)
+      compose_file_params = {
+        path: absolute_path,
+        content: Base64.encode64(compose_file_data),
+        source: absolute_path,
+      }
+
+      {
+        compose_file: compose_file_params,
+        dependencies: dependencies,
+      }
+    end
+
+    def deployment_name_valid?(deployment)
+      return false unless deployment.start_with?('deployment-')
+      return false unless deployment.split('-').size == 2
+
+      deployment_id = deployment.split('-').last
+      deployment_id.to_i.to_s == deployment_id
```

Comment:
Why this check?

Reply:
without this check, 'deployment-12s3' will count as valid